### PR TITLE
Refactor 'is_respondent_enrolled' function

### PR DESCRIPTION
**Pull request by insacuri**
# What and why?

The `is_respondent_enrolled` function had a weird return type where it could be either a boolean or a dict.  This is bad practice so this PR modifies that function and everything that uses it to only care about the value returned

# How to test?

# Trello

### CODE REVIEW

**Review by mfcrocker — APPROVED** [review]
Thanks for this, I hate that this is even a thing you can do in Python :D

File: frontstage/controllers/party_controller.py [hunk 1]
```diff
@@ -544,15 +544,13 @@ def display_button(status, ci_type):
     return not (ci_type == "EQ" and status in CLOSED_STATE)
 
 
-def is_respondent_enrolled(party_id, business_party_id, survey_short_name, return_survey=False):
-    survey = survey_controller.get_survey_by_short_name(survey_short_name)
+def is_respondent_enrolled(party_id: str, business_party_id: str, survey: dict) -> bool:
```
Comment (creativeappdevamit) [inline frontstage/controllers/party_controller.py:547]: why cant we raise an exception when it is not enrolled.

File: frontstage/views/surveys/download_survey.py [hunk 2]
```diff
@@ -25,7 +26,8 @@ def download_survey(session):
 
     # Check if respondent has permission to download for this case
     case = case_controller.get_case_by_case_id(case_id)
-    party_controller.is_respondent_enrolled(party_id, business_party_id, survey_short_name)
+    survey = survey_controller.get_survey_by_short_name(survey_short_name)
+    party_controller.is_respondent_enrolled(party_id, business_party_id, survey)
```
Comment (creativeappdevamit) [inline frontstage/views/surveys/download_survey.py:30]: same as above

File: frontstage/views/surveys/upload_survey.py [hunk 3]
```diff
@@ -38,7 +39,8 @@ def upload_survey(session):
         )
 
     # Check if respondent has permission to upload for this case
-    party_controller.is_respondent_enrolled(party_id, business_party_id, survey_short_name)
+    survey = survey_controller.get_survey_by_short_name(survey_short_name)
+    party_controller.is_respondent_enrolled(party_id, business_party_id, survey)
```
Comment (creativeappdevamit) [inline frontstage/views/surveys/upload_survey.py:43]: raising an exception?

**Pull request merged**